src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
NaN and Infinity are represented by an exponent with all bits set. Converting from Single to Double seems to preserve that situation, I've tested it with several values. Everything else wouldn't make any sense, anyway. The current test code even includes two different values for NaN to make that sure (cast_to.cs, 286, 374).

Since there is no arithmetic expression as mentioned by @JonHanna the compiler cannot "optimize" something. But we could add some additional test values for NaN, but I'm not sure if that helps.